### ISSUE TITLE
WriteableBitmap throws an exception when accessing PixelBuffer

### ISSUE THREAD

**Issue by gmgallo** [microsoft/microsoft-ui-xaml#9533]
### Describe the bug

Following the code example the WriteableBitmap is not properly costructed. PixelBuffer is null after construction. This is the control inspetion in Visual Studio after creation:

![image](https://github.com/microsoft/WindowsAppSDK/assets/11802515/f678c70f-9901-410b-b6b2-6277afb821a8)

### Steps to reproduce the bug

 ```
    var renderTarget = new WriteableBitmap(2048, 1024);

    // renderTarget is not null here, but next line throws an exeption
    var buf = renderTarget.PixelBuffer;

```

### Expected behavior

As in the example here [WriteableBitmap.PixelBuffer property](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.media.imaging.writeablebitmap.pixelbuffer?view=windows-app-sdk-1.5)

### NuGet package version

Windows App SDK 1.5.1: 1.5.240311000

### Packaging type

Packaged (MSIX)

### Windows version

Windows 10 version 22H2 (19045, 2022 Update)

### IDE

Visual Studio 2022-preview

**Comment by gmgallo** [comment]
The error throws at runtime, all looks good in the editor.   There is no other code between the two lines of code shown. The screenshot shows the object inspection with a breakpoint on the line var buf = ....
I'll try on an empty application.

**Comment by kmahone** [comment]
I was also unable to reproduce this issue. @gmgallo do you hit this in a blank app?

**Comment by gmgallo** [comment]
No, I haven't tested in a blank app yet. I have to move on and don't have time to investigate. 

I suspect that Visual Stuido is the one to blame, not this assembly. Strange things happen all the time, like [this error](https://developercommunity.visualstudio.com/t/vs2022-reporting-false-cs0103-errors-in-web-forms/1545364) is happening to me now. Xaml Hot-Reload not working on and off on the same project, and the list goes on and on,
